hack/test.sh

```diff
@@ -0,0 +1,34 @@
+#!/bin/sh
+
+set -e
+
+COVERAGE_MINIMUM=20
+COVERAGE_PASS="true"
+
+coverage run -m unittest discover src
+coverage report -m | tee coverage.log
+
+for FILE in $(find src/ -type f | grep -v -e "__" -e request_helper -e "/test" -e "src/[^/]*.py" | grep py$);
+do
+    if [ $(grep "$FILE" coverage.log | wc -l) -gt 0 ];
+    then
+        COVERAGE=$(grep "$FILE" coverage.log | awk '{print $4}' | sed 's/%//g')
+        if [ $COVERAGE -le $COVERAGE_MINIMUM ]
+        then
+            echo "FAILURE: Test coverage below target ${COVERAGE_MINIMUM}% for '$FILE'"
+            COVERAGE_PASS="false"
+        else
+            echo "SUCCESS: Test coverage at or above target ${COVERAGE_MINIMUM}% for '$FILE'"
+        fi
+    else
+        "FAILURE: Unable to find test results for '$FILE'"
+        COVERAGE_PASS="false"
+    fi
+done
+rm -f coverage.log
```

Comment:
I was OK leaving .coverage because it's generated by the tool, not stuff I added.  If there's some useful state locally I see no harm leaving it around.